lib/facter/resolvers/aix/ffi/ffi_helper.rb

```diff
@@ -1,8 +1,16 @@
 # frozen_string_literal: true
 
-require 'ffi'
 require_relative 'structs'
 require_relative 'ffi'
+
+# ffi is only a development dependency and may not be present
+begin
+  require 'ffi'
+rescue LoadError => e
+  log = Facter::Log.new(self)
+  log.debug("The ffi gem has not been installed: #{e}")
+end
+
```

Comment:
In theory, `ffi` related code is never required in the main entry point https://github.com/puppetlabs/facter/blob/4c69cb9aa08ecec4e3ab56cb47656c4c826d38dc/lib/facter/framework/core/file_loader.rb#L28-L97

Instead, we lazily load `ffi_helper`, etc such as above when trying to resolve the fact at runtime, like the AIX networking facts: https://github.com/puppetlabs/facter/blob/4c69cb9aa08ecec4e3ab56cb47656c4c826d38dc/lib/facter/resolvers/aix/networking.rb#L63 

I think I'd expect `require 'facter'` to never cause `ffi` to be loaded unless a specific fact depends on it. And if that happens, I think an `error` message should be displayed that some facts will be omitted because ffi is not installed.

There's a possible catch-22 though because facter does execute some code to figure out which OS it's on and therefore, which OS specific resolvers to load. So maybe that is triggering ffi to be loaded?

Reply:
It looks like Windows always loads the identity resolver:

https://github.com/puppetlabs/facter/blob/4c69cb9aa08ecec4e3ab56cb47656c4c826d38dc/lib/facter/framework/core/file_loader.rb#L902

Which in turn uses FFI to find usernames: 

https://github.com/puppetlabs/facter/blob/main/lib/facter/resolvers/windows/identity.rb#L19

So I can try rescuing LoadError in the `find_username` method.